R/get_inboveg_recording.R

```diff
@@ -262,6 +167,55 @@ get_inboveg_recording <- function(
   return(query_result)
 }
 
+
+
+#' @importFrom assertthat assert_that
+#' @importFrom glue glue_sql
+complete_sql_statement <- function(
+    survey_name,
+    recording_givid,
+    user_reference,
+    common_part,
+    connection) {
+  survey_where <- ifelse(
+    length(survey_name) != 1,
+    " AND ivS.Name IN ({survey_name*})",
+    " AND ivS.Name LIKE {survey_name}"
+  )
+
+  user_reference_where <- ifelse(
+    length(user_reference) != 1,
+    "ivR.UserReference IN ({user_reference*})",
+    "ivR.UserReference LIKE {user_reference}"
+  )
+
+  recording_givid_where <- ifelse(
+    length(recording_givid) != 1,
+    "ivR.[RecordingGivid] IN ({recording_givid*})",
+    "ivR.[RecordingGivid] LIKE {recording_givid}"
+  )
+
+  combined_where_clauses <-
+    paste0(
+      survey_where,
+      " AND ",
+      user_reference_where,
+      " AND ",
+      recording_givid_where
+    )
+
+  sql_statement <- glue_sql(
+    common_part,
+    combined_where_clauses,
+    survey_name = survey_name,
+    recording_givid = recording_givid,
+    user_reference = user_reference,
+    .con = connection)
+
+  return(sql_statement)
+}
+
+
```

Comment:
Could you add this function to a separate file named after the function with some explanation to make it easier for developers to find their way in the package?  When making changes in `get_inboveg_ppa()`, nobody will search for this function here.
For the documentation, just mention inboveg in the title and add explain in one sentence it is a helper function for functions xxx (with maybe some minimal explanation to understand it as a developer).  And add `@noRd` if the function is not exported.

Reply:
[391a74f](https://github.com/inbo/inbodb/pull/51/commits/391a74fe504d1ea86507ea1b7048c2bfd534d82c)